lib/Dialect/HW/HWOps.cpp

```diff
@@ -747,6 +740,13 @@ void HWModuleOp::build(OpBuilder &builder, OperationState &result,
     body->addArgument(type, loc);
   }
 
+  // Add result ports attribute.
```

Comment:
This adds 2 more iterations over the ports.  Can't this be done in the previous loop?